subprojects/packagefiles/mpdecimal/meson.build

```diff
@@ -202,66 +201,117 @@ if machine == 'auto'
+    add_project_link_arguments('-m64', language: ['c', 'cpp'])
+  endif
 
 elif machine == 'ppro'
-  add_project_arguments('-DCONFIG_32', '-DPPRO', language : 'c')
   conf.set('MPD_HEADER_CONFIG', mpd_header_config_32)
+  add_project_arguments('-DCONFIG_32', '-DPPRO', '-DASM', language: 'c')
   if host_machine.system() == 'windows'
-    add_project_arguments('/DMASM', language: 'c')
+    add_project_arguments('-DMASM', language: 'c')
+  else
+    add_project_arguments('-DASM', language: 'c')
+    add_project_arguments('-m32', language: ['c', 'cpp'])
+    add_project_link_arguments('-m32', language: ['c', 'cpp'])
+  endif
+
+  # Some versions of gcc miscompile inline asm:
+  # http://gcc.gnu.org/bugzilla/show_bug.cgi?id=46491
+  # http://gcc.gnu.org/ml/gcc/2010-11/msg00366.html
+  if cc.get_id() == 'gcc' and cc.run('''
+    __attribute__((noinline)) int
+    foo(int *p) {
+      int r;
+      asm ( "movl \$6, (%1)\n\t"
+            "xorl %0, %0\n\t"
+            : "=r" (r) : "r" (p) : "memory"
+      );
+      return r;
+    }
+    int main() {
+      int p = 8;
+      if ((foo(&p) ? : p) != 6)
+        return 1;
+      return 0;
+    }''',
+    args: ['-O2'],
+    name: 'gcc ipa-pure-const').returncode() != 0
+
+    add_project_arguments('-fno-ipa-pure-const', language: 'c')
   endif
 
 elif machine == 'ansi32'
-  add_project_arguments('-DCONFIG_32', '-DANSI', language : 'c')
   conf.set('MPD_HEADER_CONFIG', mpd_header_config_32)
+  add_project_arguments('-DCONFIG_32', '-DANSI', language: 'c')
+  if host_machine.system() != 'windows'
+    add_project_arguments('-m32', language: ['c', 'cpp'])
+    add_project_link_arguments('-m32', language: ['c', 'cpp'])
+  endif
 
 elif machine == 'ansi-legacy'
-  add_project_arguments('-DCONFIG_32', '-DANSI', '-DLEGACY_COMPILER', language : 'c')
-  conf.set('MPD_HEADER_CONFIG', mpd_header_config_32_legacy)
+  conf.set('MPD_HEADER_CONFIG', mpd_header_config_32_LEGACY)
+  add_project_arguments('-DCONFIG_32', '-DANSI', '-DLEGACY_COMPILER', language: 'c')
+  if host_machine.system() != 'windows'
+    add_project_arguments('-m32', language: ['c', 'cpp'])
+    add_project_link_arguments('-m32', language: ['c', 'cpp'])
+  endif
 
 elif machine == 'universal'
-  add_project_arguments('-DUNIVERSAL', language : 'c')
-  conf.set('MPD_HEADER_CONFIG', mpd_header_config_universal)
+  add_project_arguments('-DUNIVERSAL', language: 'c')
+  conf.set('MPD_HEADER_CONFIG', mpd_header_config_UNIVERSAL)
+  if host_machine.system() == 'windows'
+    error('machine "universal" not supported on Windows')
+  endif
+
 endif
 
-build_dll = []
-use_dll = []
+
+build_mpdec_compiler_args = []
 if host_machine.system() == 'windows'
   add_project_arguments('/wd4200', '/wd4204', '/wd4221', '/D_CRT_SECURE_NO_WARNINGS', '/nologo', language: 'c')
   if get_option('default_library') != 'static'
-    build_dll = ['/DBUILD_DLL']
-    use_dll = ['/DUSE_DLL']
+    build_mpdec_compiler_args = ['-DBUILD_LIBMPDEC']
```

Comment:
Looking at the changes upstream, it seems that there is still a `_DLL` macro guard which determines if `__declspec(dllimport)` is part of the use flags.

I think we may still want that?

Reply:
I didn't find where in `Makefile.vc` that is set, thus I didn't include it:
```
~/src/wrapdb/subprojects/mpdecimal-2.5.1$ find . -name Makefile.vc | xargs grep _DLL
~/src/wrapdb/subprojects/mpdecimal-2.5.1$
```

Reply:
The Makefile.vc doesn't provide any export interface such as pkg-config, nor does it really have usage documentation... I think that this is just left for users to blindly guess at. :(

Reply:
The examples and tests work fine without it, thus I think we don't need it.